packages/govern-console/src/components/TransactionKeeper/TransactionKeeper.tsx

```diff
@@ -35,17 +35,14 @@ const TransactionKeeper: React.FC<TransactionKeeperProps> = ({
 
   const updateTransaction = React.useCallback(
     (updatedTransaction: CustomTransaction, updatedTransactionIndex: number) => {
-      console.log('updating transaction' + updatedTransactionIndex);
-      const updatedTransactions = transactions.map((transaction, index) =>
-        index === updatedTransactionIndex ? updatedTransaction : transaction,
```

Comment:
consider using the `updater` function of `SetState` to avoid race conditions when updating array state.
https://stackoverflow.com/questions/26253351/correct-modification-of-state-arrays-in-react-js
https://reactjs.org/docs/react-component.html#setstate

Reply:
Also, can we rename `updateTransaction()` to `updateTransactionStatus(txIndex, newStatus)` and get rid of the the following code ?  This way, it's clear that we are only changing the status, nothing else in the transaction?

```
        let updatedTransaction = produce(transaction, (draft) => {
          draft.status = CustomTransactionStatus.InProgress;
        });
```

Reply:
yes make sense,
probably it would be better to address this and also there is more concern in TransactionKeeper,
i think a thorough refactoring